syz-manager/html.go

```diff
@@ -296,21 +318,41 @@ func (mgr *Manager) httpCoverFallback(w http.ResponseWriter, r *http.Request) {
 	executeTemplate(w, fallbackCoverTemplate, data)
 }
 
-func (mgr *Manager) httpFuncCover(w http.ResponseWriter, r *http.Request) {
+func (mgr *Manager) prepareCSVCover(w http.ResponseWriter) (*cover.ReportGenerator, error) {
 	if !mgr.cfg.Cover {
 		http.Error(w, "coverage is not enabled", http.StatusInternalServerError)
-		return
+		return nil, fmt.Errorf("coverage is not enabled")
 	}
 	rg, err := getReportGenerator(mgr.cfg)
 	if err != nil {
 		http.Error(w, fmt.Sprintf("failed to generate coverage profile: %v", err), http.StatusInternalServerError)
-		return
+		return nil, err
 	}
+
 	mgr.mu.Lock()
 	defer mgr.mu.Unlock()
+
```

Comment:
This lock now does not protect anything (and the same in prepareHTMLCover). So now reading of coverage is racy and can crash or anything.
I think we need to move all of this logic into httpCoverCover.
The only difference between prepareCSVCover and prepareHTMLCover is that prepareHTMLCover calls httpCoverFallback if coverage is not enabled. This can be a bool flag to  httpCoverCover.